src/std/debug.c

```diff
@@ -248,14 +250,21 @@ HL_API int hl_debug_wait( int pid, int *thread, int timeout ) {
 	return mdbg_session_wait(pid, thread, timeout);
 #	elif defined(USE_PTRACE)
 	int status;
-	int ret = waitpid(pid,&status,0);
-	//printf("WAITPID=%X %X\n",ret,status);
+	// *** HACK ***
+	// usleep here is needed for a good result.
+	// Without it, waitpid can miss many stop event;
+	// With it, and more we wait, less we miss stop event.
+	usleep(100 * 1000);
+	int ret = waitpid(pid, &status, WNOHANG);
```

Comment:
I suspect that `waitpid` with `WNOHANG` is also vulnerable to a race condition with libuv.

I'm not sure how to fix this other than by doing the same thread & semaphore dance as the Mac debugger.

Reply:
I also suspect the `waitpid` hack was for libuv (or vscode) waiting for all childs.

I tested your implementation with thread (it helps me a lot!), but it failed when we are actually waiting (by default the vscode extension call wait with a timeout = 0) and has a timeout. I didn't managed to get it works yet :'(
You said @ncannasse refused your PR due to complexity, so I tried this hack. I will do the thread & semaphore if hack is not a good option either.

Reply:
That was another PR, I never actually submitted the thread & semaphore implementation.

Reply:
Sorry I misunderstood :(